### PR TITLE
add provider register mock

### PR DESCRIPTION
**Pull request by EvenLjj**
add provider register mock

### CODE REVIEW

File: sofa-boot-project/sofa-boot-autoconfigure/src/main/java/com/alipay/sofa/boot/autoconfigure/rpc/SofaRpcAutoConfiguration.java [hunk 1]
```diff
@@ -38,11 +38,7 @@
 import com.alipay.sofa.rpc.boot.health.RpcAfterHealthCheckCallback;
 import com.alipay.sofa.rpc.boot.runtime.adapter.helper.ConsumerConfigHelper;
 import com.alipay.sofa.rpc.boot.runtime.adapter.helper.ProviderConfigHelper;
-import com.alipay.sofa.rpc.boot.runtime.adapter.processor.ConsumerConfigProcessor;
-import com.alipay.sofa.rpc.boot.runtime.adapter.processor.ConsumerMockProcessor;
-import com.alipay.sofa.rpc.boot.runtime.adapter.processor.DynamicConfigProcessor;
-import com.alipay.sofa.rpc.boot.runtime.adapter.processor.ProcessorContainer;
-import com.alipay.sofa.rpc.boot.runtime.adapter.processor.ProviderConfigProcessor;
+import com.alipay.sofa.rpc.boot.runtime.adapter.processor.*;
```
Comment (OrezzerO) [inline sofa-boot-project/sofa-boot-autoconfigure/src/main/java/com/alipay/sofa/boot/autoconfigure/rpc/SofaRpcAutoConfiguration.java:41]: do not use * at import statement

File: sofa-boot-project/sofa-boot-core/rpc-sofa-boot/src/main/java/com/alipay/sofa/rpc/boot/config/SofaBootRpcConfigKeys.java [hunk 2]
```diff
@@ -25,9 +25,17 @@
...
+                                                             "", false, "默认注册中心实现",
+                                                             new String[] { "default.registry" });
+
+    public static ConfigKey<String> DISABLE_REGISTER_PUB = ConfigKey
+                                                             .build(
+                                                                 "sofa.rpc.registry.disablePub",
+                                                                 "",
+                                                                 false,
+                                                                 "服务提供方不注册开关",
+                                                                 new String[] { "disable_confreg_pub" });
```
Comment (OrezzerO) [inline sofa-boot-project/sofa-boot-core/rpc-sofa-boot/src/main/java/com/alipay/sofa/rpc/boot/config/SofaBootRpcConfigKeys.java:39]: should be `disable_confireg_pub`

File: sofa-boot-project/sofa-boot-core/rpc-sofa-boot/src/test/java/com/alipay/sofa/rpc/boot/test/runtime/adapter/processor/ProviderRegisterProcessorTest.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    @Test
+    public void test() {
+        ProviderConfig providerConfig = new ProviderConfig();
+        providerConfig.setRegister(true);
+        providerRegisterProcessor.processorProvider(providerConfig);
+        Assert.assertTrue(providerConfig.isRegister());
+
+        System.setProperty("sofa.rpc.registry.disablePub", "true");
+        providerRegisterProcessor.processorProvider(providerConfig);
+        Assert.assertFalse(providerConfig.isRegister());
```
Comment (OrezzerO) [inline sofa-boot-project/sofa-boot-core/rpc-sofa-boot/src/test/java/com/alipay/sofa/rpc/boot/test/runtime/adapter/processor/ProviderRegisterProcessorTest.java:40]: System properties should roll back

File: sofa-boot-project/sofa-boot-core/rpc-sofa-boot/src/main/java/com/alipay/sofa/rpc/boot/runtime/adapter/processor/ProviderRegisterProcessor.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+package com.alipay.sofa.rpc.boot.runtime.adapter.processor;
+
+import com.alipay.sofa.common.config.SofaConfigs;
+import com.alipay.sofa.rpc.boot.config.SofaBootRpcConfigKeys;
+import com.alipay.sofa.rpc.config.ProviderConfig;
+
+/**
+ * @author BaoYi
+ */
+public class ProviderRegisterProcessor implements ProviderConfigProcessor {
```
Comment (OrezzerO) [inline sofa-boot-project/sofa-boot-core/rpc-sofa-boot/src/main/java/com/alipay/sofa/rpc/boot/runtime/adapter/processor/ProviderRegisterProcessor.java:26]: class name can not indicate the function of this class

**Review by OrezzerO — CHANGES_REQUESTED** [review]

**Review by OrezzerO — APPROVED** [review]
LGTM

**Comment by alaneuler** [comment]
fix: https://github.com/sofastack/sofa-boot/issues/859

**Pull request merged**